src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+import styles from './CheckboxWithTextfields.scss';
+
+import ImageButton from 'src/shared/image-button/ImageButton';
+import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
+import PasteOrUpload from '../paste-or-upload/PasteOrUpload';
+
+export type CheckboxWithTextfieldsProps = {
+  values: string[];
+  label: string;
+  allowMultiple: boolean;
+  disabled?: boolean;
+  onChange: (values: string[]) => void;
+};
+
+const CheckboxWithTextfields = (props: CheckboxWithTextfieldsProps) => {
+  const [isChecked, setIsChecked] = useState(false);
+  const [values, setValues] = useState<(string | null)[]>(props.values);
```

Comment:
Is there any reason to have these values in local state, when they are updated from props, and any change in the input calls onChange in the props?

Reply:
This is just to implement the option to display a new input field when the add button is clicked, by pushing `null` to the values.